code/rounds/InProgressRound.cs

```diff
@@ -160,12 +142,13 @@ private TTTTeam IsRoundOver()
 
         private void AssignRoles()
         {
-            // TODO: There should be a neater way to handle this logic.
-            int traitorCount = (int) Math.Max(_startingRoundPlayers.Count * 0.25f, 1f);
+            List<TTTPlayer> alivePlayers = Utils.GetAlivePlayers();
+
+            int traitorCount = (int) Math.Max(alivePlayers.Count * 0.25f, 1f);
 
             for (int i = 0; i < traitorCount; i++)
             {
-                List<TTTPlayer> unassignedPlayers = _startingRoundPlayers.Where(p => p.Role is NoneRole).ToList();
+                List<TTTPlayer> unassignedPlayers = alivePlayers.Where(p => p?.Role is NoneRole).ToList();
```

Comment:
Did you tested this? Are forced spectators alive? And why `p?`

Reply:
Removed `?`, yeah I've been doing some intensive testing making sure this large logic change doesn't mess things up.